velox/functions/FunctionRegistry.cpp

```diff
@@ -141,4 +141,96 @@ resolveVectorFunctionWithMetadata(
   return exec::resolveVectorFunctionWithMetadata(functionName, argTypes);
 }
 
+bool isCompanionFunctionName(
+    const std::string& name,
+    const std::unordered_map<std::string, exec::AggregateFunctionEntry>&
+        aggregateFunctions) {
+  auto suffixOffset = name.rfind("_partial");
+  if (suffixOffset == std::string::npos) {
+    suffixOffset = name.rfind("_merge_extract");
+  }
```

Comment:
Can we avoid so many checks of suffixOffset before looking into aggregateFunctions ? It might be better to do the lookup as soon as we get a suffixOffset != std::string::npos instead.